modules/light-clients/08-wasm/types/querier.go

```diff
@@ -0,0 +1,150 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
+)
+
+var (
+	_ wasmvmtypes.Querier = (*DefaultQuerier)(nil)
+
+	queryPlugins = NewDefaultQueryPlugins()
+)
+
+type DefaultQuerier struct {
```

Comment:
Querier is a part of the public api now as the users can extend functionality.  I think this should be renamed to `QueryHandler` now that users cannot change the entire querier. Moving it to `ibcwasm` is problematic because the gas_register is in types.

Reply:
Yes, but it feels weird to have a public interface whose consumer is private. I can change this if others agree

Reply:
this is private now